components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -563,6 +561,48 @@ public Map<String, Object> mapExistingOAuthClient(String jsonString, String user
             throw new APIManagementException("Key Manager " + keyManagerName + " doesn't support to generate" +
                     " Client Application", ExceptionCodes.KEY_MANAGER_NOT_SUPPORT_OAUTH_APP_CREATION);
         }
+
+        //Get application ID
+        int applicationId = apiMgtDAO.getOrgSpecificApplicationId(applicationName, userName, tenantDomain);
+
+        // Checking if clientId is mapped with another application.
+        if (apiMgtDAO.isKeyMappingExistsForConsumerKeyOrApplication(applicationId, keyManagerName, keyManagerId,
```

Comment:
Shall we fix isKeyMappingExistsForConsumerKeyOrApplication method too, to take the application UUID and organization instead of taking the autogenerated ID?

Reply:
We decided to keep it as it is because application key mapping table maps application using the integer ID. Otherwise we would have to do a join in the query.